deepchem/models/gbdt_models/gbdt_model.py

```diff
@@ -107,16 +120,19 @@ def fit(self, dataset: Dataset):
                                                             test_size=0.2,
                                                             random_state=seed,
                                                             stratify=stratify)
-        self.model.fit(X_train,
-                       y_train,
-                       early_stopping_rounds=self.early_stopping_rounds,
-                       eval_metric=self.eval_metric,
-                       eval_set=[(X_test, y_test)])
+        self.model.fit(
+            X_train,
+            y_train,
+            callbacks=[self.callbacks],
+            eval_metric=self.eval_metric,
+            eval_set=[(X_test, y_test)])
 
         # retrain model to whole data using best n_estimators * 1.25
         if self.model.__class__.__name__.startswith('XGB'):
+            assert self.model.best_iteration<self.model.n_estimators
```

Comment:
@rbharath Added a test to check if the best iteration is less than the number of estimators. If early stopping is triggered, this condition would be true.